### DISCUSSION TITLE
Credentials for importing from repos with secured webdav remotes

### DISCUSSION THREAD

**Discussion by mdekstrand** [treeverse/dvc#7869] [Help] [answered]
I am trying to use `import` to import files from a repository that uses password-protected WebDAV as its remote, with the password in `.dvc/config.local` when working with that repository on its own.

However, when I try to import into another repository, it needs the WebDAV credentials. Is there any way to make them available short of committing them to the source repository? I have tried creating a remote in the destination repository with the same name, but that did not work.

For S3 remotes, there is no problem because the credentials are stored in `~/.aws/credentials`.

**Comment by skshetry** [comment, marked as answer]
You can [keep them in your global config](https://dvc.org/doc/command-reference/config#--global).

```console
dvc remote modify --global remote user $SECRET_USER
dvc remote modify --global remote password $SECRET_KEY
```

WebdavFileSystem also supports [`netrc`](https://www.python-httpx.org/advanced/#netrc-support).

eg:
```console
# ~/.netrc should only be accessible to the current user
(umask 077; echo "machine $SECRET_HOSTNAME login $SECRET_USER password $SECRET_KEY" > ~/.netrc)
```

You may also be interested in https://github.com/iterative/dvc/issues/6416.

**Reply by mdekstrand** [reply]
Ooh, `~/.netrc` worked and is a good solution. Now I'm stuck on another webdav remote bug (my remote is `wevdavs://host/dav/bookdata`, and it's trying to look at `/dav/bookdata/dav/bookdata/??/name`), but authentication seems to be working.
